bugtool/cmd/configuration.go

```diff
@@ -0,0 +1,247 @@
+		{"/proc/config", fmt.Sprintf("%s/kernel-config", confDir)},
+		{"/proc/config.gz", fmt.Sprintf("%s/kernel-config.gz", confDir)},
+	}
+
+	if len(k8sPods) == 0 {
+		kernel, _ := execCommand("uname", "-r")
+		kernel = strings.TrimSpace(kernel)
+		l := Location{fmt.Sprintf("/boot/config-%s", kernel),
+			fmt.Sprintf("%s/kernel-config-%s", confDir, kernel)}
+		locations = append(locations, l)
+
+		for _, location := range locations {
+			if _, err := os.Stat(location.Src); os.IsNotExist(err) {
+				continue
+			}
+			commands = append(commands, fmt.Sprintf("cp %s %s", location.Src, location.Dst))
+		}
+	} else {
+		for _, pod := range k8sPods {
+			prompt := podPrefix(pod, "uname -r")
+			cmd, args := split(prompt)
+			kernel, _ := execCommand(cmd, args...)
+			kernel = strings.TrimSpace(kernel)
+			l := Location{fmt.Sprintf("/boot/config-%s", kernel),
+				fmt.Sprintf("%s/kernel-config-%s", confDir, kernel)}
+			locations = append(locations, l)
+
+			for _, location := range locations {
+				kubectlArg := fmt.Sprintf("%s/%s:%s", k8sNamespace, pod, location.Src)
+				cmd := fmt.Sprintf("%s %s %s %s", "kubectl", "cp", kubectlArg, location.Dst)
+				commands = append(commands, cmd)
+			}
+		}
+	}
+	return commands
+}
+
+func copyCiliumInfoCommands(ciliumDir string, k8sPods []string) []string {
+	// Most of the output should come via debuginfo but also adding
+	// these ones for skimming purposes
+	ciliumCommands := []string{
+		"cilium debuginfo",
+		"cilium config",
+		"cilium bpf tunnel list",
+		"cilium bpf lb list",
+		"cilium bpf endpoint list",
+		"cilium bpf ct list global",
+		"cilium status",
+	}
+	var commands []string
+
+	stateDir := filepath.Join(defaults.RuntimePath, defaults.StateDir)
+	if len(k8sPods) == 0 { // Assuming this is a non k8s deployment
+		dst := filepath.Join(ciliumDir, defaults.StateDir)
+		commands = append(commands, fmt.Sprintf("cp -r %s %s", stateDir, dst))
+		for _, cmd := range ciliumCommands {
+			// Add the host flag if set
+			if len(host) > 0 {
+				cmd = fmt.Sprintf("%s -H %s", cmd, host)
+			}
+			commands = append(commands, cmd)
+		}
+	} else { // Found k8s pods
+		for _, pod := range k8sPods {
+			dst := filepath.Join(ciliumDir, fmt.Sprintf("%s-%s", pod, defaults.StateDir))
+			kubectlArg := fmt.Sprintf("%s/%s:%s", k8sNamespace, pod, stateDir)
+			// kubectl cp kube-system/cilium-xrzwr:/var/run/cilium/state cilium-xrzwr-state
+			if _, err := execCommand("kubectl", "cp", kubectlArg, dst); err != nil {
+				fmt.Fprintf(os.Stderr, "Could not copy state directory %s\n", err)
+			}
+
+			for _, cmd := range ciliumCommands {
+				// Add the host flag if set
+				if len(host) > 0 {
+					cmd = fmt.Sprintf("%s -H %s", cmd, host)
+				}
+				commands = append(commands, podPrefix(pod, cmd))
+			}
+		}
+	}
+
+	return commands
+}
+
+func k8sCommands(allCommands []string, pods []string) []string {
+	var commands = []string{
+		"kubectl describe nodes",
+		"kubectl version",
+		"kubectl -n kube-system get pods",
```

Comment:
you don't need to run this since command you have wrote a more general command on line 226

Reply:
ah, thanks.